src/main/java/org/cadixdev/mercury/mixin/MixinRemapperVisitor.java

```diff
@@ -364,4 +365,59 @@ private static MethodSignature convertSignature(final String name, final IMethod
         return new MethodSignature(name, new MethodDescriptor(parameters, convertType(binding.getReturnType())));
     }
 
+    private void remapPrivateMixinTarget(
+        final AST ast, final TypeDeclaration typeDeclaration, final ITypeBinding typeBinding
+    ) {
+        IAnnotationBinding[] annotations = typeBinding.getAnnotations();
+        List modifiers = typeDeclaration.modifiers();
+        for (Object modifier : modifiers) {
+            if (modifier instanceof NormalAnnotation) {
+                NormalAnnotation normalAnnotation = (NormalAnnotation) modifier;
+                for (Object raw : normalAnnotation.values()) {
+                    MemberValuePair pair = (MemberValuePair) raw;
+                    if ("targets".equals(pair.getName().getIdentifier())) {
+                        Expression targets = pair.getValue();
+                        if(targets instanceof StringLiteral){
+                            remapPrivateMixinTargetLiteral(ast, ((StringLiteral) targets));
+                        }
+                        else if (targets instanceof ArrayInitializer) {
+                            List expressions = ((ArrayInitializer) targets).expressions();
+                            for (Object expression : expressions) {
+                                if (expression instanceof StringLiteral) {
+                                    remapPrivateMixinTargetLiteral(ast, (StringLiteral) expression);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private void remapPrivateMixinTargetLiteral(AST ast, StringLiteral literal) {
+        String className = literal.getLiteralValue().replace('.', '/');
+        if (!className.isEmpty()) {
+            ClassMapping<?, ?> classMapping = mappings.getTopLevelClassMapping(className)
```

Comment:
This is going to fail to remap inner classes, this should be `getClassMapping(String)`.

cc @phase as I think there was an earlier discussion on this

Reply:
`getClassMappings` failed to remap mixins targeting anonymous inner classes that are mapped to top level classes, like `CL net/minecraft/network/NetworkManager$2 el` was being remapped to `ej$2`.

Reply:
I'd consider mappings in that form unlikely and unfortunate personally, they're not worth supporting at the cost of supporting `CL net/minecraft/network/NetworkManager$2 net/minecraft/server/network/NetworkManager$2`.

Given that `net/minecraft/network/NetworkManager$2` shouldn't register as a top-level class anyhow - this shouldn't work (for that purpose).

Reply:
Doing something like `getTopLevelClassMapping(className).orElseGet(computeClassMappings(className)).orElse(null)` should support all cases of mappings.